project_recalculate/models/project_project.py

```diff
@@ -0,0 +1,79 @@
+# -*- coding: utf-8 -*-
+##############################################################################
+# For copyright and license notices, see __openerp__.py file in root directory
+##############################################################################
+
+from openerp import models, fields, api, _
+from openerp.exceptions import Warning
+
+
+class ProjectProject(models.Model):
+    _inherit = 'project.project'
+
+    calculation_type = fields.Selection(
+        [('none', 'No calculation'),
+         ('date_begin', 'Date begin'),
+         ('date_end', 'Date end')],
+        string='Calculation type', default='date_begin', required=True,
+        help='How to calculate tasks with date start or date end references')
+
+    def _dates_prepare(self):
```

Comment:
I suggest renaming this to `_calc_start_end_dates`
IMO it would be clearer on what it does.